huaweicloud/services/modelarts/resource_huaweicloud_modelarts_service.go

```diff
@@ -0,0 +1,1007 @@
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func ResourceModelartsService() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceModelartsServiceCreate,
+		UpdateContext: resourceModelartsServiceUpdate,
+		ReadContext:   resourceModelartsServiceRead,
+		DeleteContext: resourceModelartsServiceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Update: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Service name, which consists of 1 to 64 characters.`,
+			},
+			"infer_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Inference mode.`,
+			},
+			"config": {
+				Type:     schema.TypeList,
+				Elem:     modelartsServiceConfigSchema(),
+				Required: true,
+			},
+			"workspace_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `ID of the workspace to which a service belongs.`,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The description of the service.`,
+			},
+			"pool_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `Specifies the ID of the new dedicated resource pool.`,
+			},
+			"vpc_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The VPC ID to which a real-time service instance is deployed.`,
+			},
+			"subnet_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The subnet ID.`,
+			},
+			"security_group_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The security group ID.`,
+			},
+			"schedule": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Elem:     modelartsServiceScheduleSchema(),
+				Optional: true,
+				Computed: true,
+			},
+			"additional_properties": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Elem:     modelartsServiceAdditionalPropertySchema(),
+				Optional: true,
+				Computed: true,
+			},
+			"change_status_to": {
```

Comment:
why not use `status` directly?

Reply:
The service may be stopped after running for some hours if set `schedule`.
So a separate field is better.